File: 1_C#(VisualStudio)/1_workspace_ASP.NETCoreMVC/1_書籍検索アプリ_移植元Java/Models/BookDataListDAO.cs
using MySql.Data.MySqlClient;
using Mysqlx.Crud;
using 書籍検索アプリ_移植元Java.Models.beans;

namespace 書籍検索アプリ_移植元Java.Models
{
   public class BookDataListDAO {
	
		//DAO
		private const string FIND_ALL_SQL =
			"SELECT * FROM bookdatalistdb";

		private const string FIND_BY_AUTHOR_SQL =
			"SELECT * FROM bookdatalistdb WHERE author = @author";


		//----------
		//コンストラクタ
		public BookDataListDAO() {
		}
	
		// 全件取得
        public static List<BookDataBean> FindAll()
        {
            return QueryExecute(FIND_ALL_SQL);
        }

		// 著者で検索
        public static List<BookDataBean> FindByAuthor(string author)
        {
            return QueryExecute(FIND_BY_AUTHOR_SQL, ("@author", author));
        }
	
		// 共通クエリ実行
        private static List<BookDataBean> QueryExecute(string sql, params (string name, object value)[] parameters)
        {
            var list = new List<BookDataBean>();

            try
            {
                //DB接続
			    using MySqlCommand cmd = new DatabaseConnect().GetConnection(sql);

                // パラメータ設定
                foreach (var (name, value) in parameters)
                {
                    cmd.Parameters.AddWithValue(name, value ?? DBNull.Value);
                }

                using var reader = cmd.ExecuteReader();

                while (reader.Read())
                {
                    list.Add(MapRow(reader));
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                throw;
            }

            return list;
        }


        // 1行 → BookDataBean に変換
        private static BookDataBean MapRow(MySqlDataReader rs) => new(

            rs["fictitiousISBN"]?.ToString() ?? "",
            rs["bookName"]?.ToString() ?? "",
            rs["author"]?.ToString() ?? "",
            rs["publicationDate"]?.ToString() ?? "",
            rs["issuer"]?.ToString() ?? "",
            rs["publisher"]?.ToString() ?? "",
            rs["printingShop"]?.ToString() ?? ""
        );
    }
}

File: 3_Java(eclipse)/1_workspace_ServletAndJSP/1_書籍検索アプリ/src/main/java/todo/model/BookDataListDAO.java
package todo.model;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.SQLTimeoutException;
import java.util.ArrayList;
import java.util.List;

import todo.model.beans.BookDataBean;

public class BookDataListDAO {
	
	//DAO
	private static final String FIND_ALL_SQL =
        "SELECT * FROM bookdatalistdb";

    private static final String FIND_BY_AUTHOR_SQL =
        "SELECT * FROM bookdatalistdb WHERE author = ?";

	//----------
	//コンストラクタ
	public BookDataListDAO() {
	}
	
	//---------
	//接続テスト用
	public static void main(String[] args) {

		var a = findAll();
		System.out.println(a.get(0).getBookName());
		System.out.println(a.get(0).getFictitiousISBN());
		System.out.println(a.get(0).getPrintingShop());
	}
	
	public static List<BookDataBean> findAll() {
		
	    return queryExecute(FIND_ALL_SQL);
	}

	public static List<BookDataBean> findByAuthor(String author) {
		
	    return queryExecute(FIND_BY_AUTHOR_SQL, author);
	}
	
	private static List<BookDataBean> queryExecute(String queryStr, Object... params) {
		
		List<BookDataBean> list = new ArrayList<>();

		try(//DB接続
			Connection conn = new DatabaseConnect().getConnection();
				
			// SQL文の実行
			PreparedStatement pstmt = conn.prepareStatement(queryStr);) {
			
			// パラメータをセット
	        for (int i = 0; i < params.length; i++) {
	        	
	            pstmt.setObject(i + 1, params[i]);
	        }

	        try (ResultSet rs = pstmt.executeQuery()) {
	        	
	            while (rs.next()) {
	                list.add(mapRow(rs));
	            }
	        }
		}catch (SQLTimeoutException e) {
			
			e.printStackTrace();
			throw new RuntimeException(e);
		}catch (SQLException e) {
			
			e.printStackTrace();
			throw new RuntimeException(e);
		}
		return list;
	}
	
	private static BookDataBean mapRow(ResultSet rs) throws SQLException {
		
        return new BookDataBean(
            rs.getString("fictitiousISBN"),
            rs.getString("bookName"),
            rs.getString("author"),
            rs.getString("publicationDate"),
            rs.getString("issuer"),
            rs.getString("publisher"),
            rs.getString("printingShop")
        );
    }
}